src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java

```diff
@@ -268,26 +236,60 @@ private static RuleSet loadRuleSet( String serverId, Settings settings, WagonMan
                                         Log logger )
         throws MojoExecutionException
     {
+        RuleSet loadedRules = isClasspathUri( rulesUri )
+                ? getRulesFromClasspath( rulesUri, logger )
+                : getRulesViaWagon( rulesUri, logger, serverId, serverId, wagonManager,
+                settings );
+
         RuleSet ruleSet = new RuleSet();
-        boolean rulesUriGiven = isRulesUriNotBlank( rulesUri );
+        ruleSet.setIgnoreVersions( loadedRules.getIgnoreVersions() );
+        ruleSet.setRules( loadedRules.getRules() );
+        return ruleSet;
```

Comment:
Why not return loadedRules ... 
missing `ruleSet.setComparisonMethod(...)`

How we can be sure that all fields are cloned, eg when somebody extends `RuleSet` ...

Reply:
Indeed. Not needed at all and the whole method became just one return statement used only in the builder, so I removed it.